gst/mqtt/mqttsrc.c

```diff
@@ -437,7 +439,14 @@ gst_mqtt_src_change_state (GstElement * element, GstStateChange transition)
       diff = GST_CLOCK_DIFF (base_time, cur_time);
       self->base_time_epoch =
           g_get_real_time () * GST_US_TO_NS_MULTIPLIER - diff;
-      GST_INFO_OBJECT (self, "GST_STATE_CHANGE_PAUSED_TO_PLAYING");
+
+      /** This handles the case when the state is changed to PLAYING again */
+      if (self->is_connected && !_do_subscribe (self)) {
+        GST_ERROR_OBJECT (self, "Failed to re-subscribe to %s",
+            self->mqtt_topic);
```

Comment:
sub/unsub methods don't look symmetric and touch different properties. Can you improve it?

1. this uses is_connected while the other uses is_subscribed
2. this uses an abstracted func, "_do_subsribe", while the other uses MQTT API directly.
3. this do not update any properties directly, while the other do so.

Recommendation: use the same property to check status and make _do_unsubscribe that is symmetric to _do_subscribe.

Not using mutex in this (while the does does) is also a bit itchy.